src/Extension.php

```diff
@@ -443,11 +445,15 @@ public function subscription_status_update( Subscription $pronamic_subscription
 
 		$status = MembershipStatus::transform_from_pronamic( $pronamic_subscription->get_status() );
 
-		if ( null === $status ) {
-			return;
+		if ( null !== $status ) {
+			$rcp_membership->set_status( $status );
 		}
 
-		$rcp_membership->set_status( $status );
+		$id = $pronamic_subscription->get_id();
+
+		if ( null !== $id ) {
+			$rcp_membership->set_gateway_subscription_id( (string) $id );
```

Comment:
Discussed with @remcotolsma at office, setting the gateway subscription ID is required to show the correct messages when editing a membership.